Fix patches!

rainwoodman opened the pull request:
![index](https://cloud.githubusercontent.com/assets/138060/13862104/b355702a-ec56-11e5-8aa7-435b5bb83df6.png)

rainwoodman committed:
Wrap paths as a whole.

rainwoodman committed:
assert xlim encloses ra_0

pmelchior commented:
That looks a lot better. Do you understand why some patches are lost on one side?

rainwoodman committed:
Add histmap and mapshow methods.

histmap creates a healpix histogram from ra, dec, and optionally weight,
then shows with mapshow.

rainwoodman commented:
I believe it is related to I didn't duplicate those vertices crossing the boundary. Ideally we can create a path with two disjoint regions, one per side, -- but I am not sure if matplotlib will be able to correctly fill them -- it probably should.

The latest commit adds two convenient functions: histmap and mapshow to the axes object:

a.histmap(ra, dec, weights=FC, nside=64, mean=True, lw=0, edgecolor='none')

rainwoodman commented:
or for an existing map:

a.mapshow(m, mask, ....)
where mask decides whether to show the pixel or not.

pmelchior commented:
OK, great. Two requests:
1. Move the functionality for histmap and mapshow into a free function, with a thin wrapper for the axes class method.
2. I'm a bit puzzled about mapshow. See my note there.

pmelchior commented on aea_projection.py:
```diff
@@ -337,6 +338,28 @@ def set_ylim(self, *args, **kwargs):
+
+        return w, mask, self.mapshow(w, mask, nest=False, **kwargs)
+
+    def mapshow(self, map, mask, nest=False, **kwargs):
```
this method does not seem to be using `map` at all !? I might be missing something, but that doesn't look right

rainwoodman replied:
```
coll = PolyCollection(v, array=map[mask], transform=self.transData, **kwargs)
~~~~~~~~~~~~~~~~~~~~~~~~~~ ++++++~~~~~~~~~~
```

pmelchior replied:
I was blind, sorry.

rainwoodman commented:
I am not so sure about '1'. 

These two function are already relatively thin. The real logic is in histogrammap() and _boundary(). 
Is it the idea we will add the other projection and reuse them? Maybe we want to have them share the same base class instead.

pmelchior commented:
I think it's a distinct possibility that we want to another projection. I've done the Lambert Conformal conic already in my old scheme, and the changes are minimal compared to the AEA.
So, I think it would be good to avoid duplication of logic that does not depend on specifics of the transformation.

Having a base class for them would be useful. Short of that, you can move the actual logic out into a free method as you have already done with `histogrammap()` and `_boundary()`.

rainwoodman committed:
Refactor-> add baseclass SkymapperAxes.

The cone transformation can be added by subclassing skymapperaxes
and overriding get_projection_class() method.

rainwoodman committed:
Use Axes coordinate when updating the clippath(.patch).

This fixes #5.

rainwoodman committed:
some preliminary support for dragging.

Trying to mimic http://bl.ocks.org/mbostock/3788999

But this code doesn't work! the data points don't move!
